### ISSUE TITLE
npm install is flaky related to "nx" "post-install" native-bindings [includes full repo]

### ISSUE THREAD

**Issue by bhouston** [nrwl/nx#26987]
### Current Behavior

I am running into an issue that 50% of my fresh npm installs are failing because of a problem with NX.

The exact error is:

```
bhouston@Ben-MacBook-Air-M3 nx-install-failure % npm install  
npm error code 1
npm error path /Users/bhouston/Coding/nx-install-failure/node_modules/lerna/node_modules/nx
npm error command failed
npm error command sh -c node ./bin/post-install
npm error /Users/bhouston/Coding/nx-install-failure/node_modules/lerna/node_modules/nx/src/native/native-bindings.js:244
npm error     throw loadError
npm error     ^
npm error
npm error Error: dlopen(/Users/bhouston/Coding/nx-install-failure/.nx/cache/18.3.5-nx.darwin-arm64.node, 0x0001): tried: '/Users/bhouston/Coding/nx-install-failure/.nx/cache/18.3.5-nx.darwin-arm64.node' (code signature invalid in <2ED4CDDA-275E-30B4-ACF0-E00AC37DB0B7> '/Users/bhouston/Coding/nx-install-failure/.nx/cache/18.3.5-nx.darwin-arm64.node' (errno=85) sliceOffset=0x00000000, codeBlobOffset=0x005E3690, codeBlobSize=0x0000BCF8), '/System/Volumes/Preboot/Cryptexes/OS/Users/bhouston/Coding/nx-install-failure/.nx/cache/18.3.5-nx.darwin-arm64.node' (no such file), '/Users/bhouston/Coding/nx-install-failure/.nx/cache/18.3.5-nx.darwin-arm64.node' (code signature invalid in <2ED4CDDA-275E-30B4-ACF0-E00AC37DB0B7> '/Users/bhouston/Coding/nx-install-failure/.nx/cache/18.3.5-nx.darwin-arm64.node' (errno=85) sliceOffset=0x00000000, codeBlobOffset=0x005E3690, codeBlobSize=0x0000BCF8)
npm error     at Module._extensions..node (node:internal/modules/cjs/loader:1544:18)
npm error     at Module.load (node:internal/modules/cjs/loader:1249:32)
npm error     at Module._load (node:internal/modules/cjs/loader:1065:12)
npm error     at Module._load (/Users/bhouston/Coding/nx-install-failure/node_modules/lerna/node_modules/nx/src/native/index.js:58:27)
npm error     at Module.require (node:internal/modules/cjs/loader:1271:19)
npm error     at require (node:internal/modules/helpers:123:16)
npm error     at Object.<anonymous> (/Users/bhouston/Coding/nx-install-failure/node_modules/lerna/node_modules/nx/src/native/native-bindings.js:135:29)
npm error     at Module._compile (node:internal/modules/cjs/loader:1434:14)
npm error     at Module._extensions..js (node:internal/modules/cjs/loader:1518:10)
npm error     at Module.load (node:internal/modules/cjs/loader:1249:32) {
npm error   code: 'ERR_DLOPEN_FAILED'
npm error }
npm error
npm error Node.js v22.2.0
```

It is occurring on MacOS and Linux.

If I run ```npm install``` a second time, it will work.

### Expected Behavior

It should not fail to run "npm install" the first time and require it to be run a second time regularly.

### Steps to Reproduce

I've got a reproduction here, just check out this repository:

https://github.com/bhouston/nx-install-failure

And the run these two commands over and over again until you get a failure.  On my current machine I get a failure about 50% of the time.

```sh
# clear previous install 
npm run clean
# do a fresh install
npm install
```

And if it does fail, try running ```npm install``` a second time and it will work.

Remember that I am on a Mac with the latest OS (beta 15, the latest) using node 22.2.0.  If you have a different OS or node version, your results may differ.

I find this error is also occurring in Linux VMs that Github Actions uses.  Thus I believe it is not related to MacOS beta.

### Nx Report

```shell
nx isn't getting installed.
```

### Package Manager Version

npm 10.7.0

### Operating System

- [X] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Seems to be related to:

https://github.com/nrwl/nx/issues/20755

https://github.com/nrwl/nx/issues/26760

**Comment by bhouston** [comment]
I have identified a fix, I upgraded lerna in the root package.json to "8.1.6".  I do not think that should have made a difference but it did.

https://github.com/lerna/lerna/releases

I think that lerna was trying to use nx 18, while in my packages.json file, it was specifying nx 19.  It appears that somehow the post-install conditions of these two separate packages have a race condition or something?  Now that lerna is using the same nx version as the root packages.json, this error is no longer occurring.

**Comment by MaxKless** [comment]
yeah it was a version mismatch of some kind. You can see in the error message that it's looking for `18.3.5-nx.darwin-arm64.node` even though you have nx 19 installed....
 But lerna `8.1.6` itself uses nx 19 so I'm not sure where this came from :/ Maybe some kind of cache?

**Comment by bhouston** [comment]
> But lerna 8.1.6 itself uses nx 19 so I'm not sure where this came from :/ Maybe some kind of cache?

Upgrading to lerna 8.1.6 works because it uses nx 19.  When I was previously using 8.1.4, it was failing because lerna at that version uses nx 18.

1. NPM should allow for multiple nx installs of different versions (e.,g. nx 18, nx 19) within the same monorepository without breaking.  This is a feature of npm.  The fact that NX breaks when this happens in this weird way suggests that something about the NX install process is buggy.

2. A lot of people have problems with the post-install step - I read this multiple times in these Github Issues.  Given the frequency with which the "cache" is brought up as being the problem, I think that nx has a broken cache system.  This shouldn't be happening with the frequency it is happening at.

Basically post-install process seems to have some fundamental bugs that are having a lot of weird effects.

We can close this, as I got around it, but fixing the fundamental issues with NX here would cut down on the frustrations and issues being filed here.
